### PR TITLE
changed static methods to normal, created interface for ResourceManager for future DI usage

### PR DESCRIPTION
**Pull request by michalwisniowski**
changed static methods to normal, created interface for ResourceManager for future DI usage

### CODE REVIEW

File: Src/Game.NET/Model/ShaderFactory.cs [hunk 1]
```diff
@@ -7,6 +7,13 @@ namespace Game.NET
 {
     public class ShaderFactory
     {
+        private readonly IResourceManager _resourceManager;
+
+        public ShaderFactory()
+        {
+            _resourceManager = new ResourceManager();
```
Comment (lukasz-pyrzyk) [inline Src/Game.NET/Model/ShaderFactory.cs]: Why field is being initialized from ctor? You can do it from one line

**Comment by michalwisniowski** [comment]
Why would you need two types of ctors ?

**Comment by lukasz-pyrzyk** [comment]
You would be able to inject this ctor by some DI container in future :)

**Comment by michalwisniowski** [comment]
Aaaa ok :) Good point. I will change it later and make na push.

**Pull request merged**